core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -119,15 +119,18 @@ angular.module('oppia').factory('QuestionObjectFactory', [
         }
       });
       if (pendingMisconceptionNamesToTag.length > 0) {
-        var returnString = 'Click on (or create) an answer ' +
-          'that is neither marked correct nor is a default answer (marked ' +
-          'above as [All other answers]) and tag the following misconceptions' +
-          ' to that answer group:';
+        var returnString = 'Remaining misconceptions that need to be ' +
+          'addressed:';
+        var misconceptionsUnorderedListElement = document.createElement('ul');
         pendingMisconceptionNamesToTag.forEach(function(misconceptionName) {
-          returnString = returnString + ' ' + misconceptionName + ',';
+          var misconceptionNameListElement = document.createElement('li');
+          misconceptionNameListElement.appendChild(
+            document.createTextNode(misconceptionName));
+          misconceptionsUnorderedListElement.appendChild(
+            misconceptionNameListElement);
         });
-        returnString = returnString.slice(0, -1);
-        return returnString;
+        return (
+          `${returnString} ${misconceptionsUnorderedListElement.outerHTML}`);
```

Comment:
Ideally don't do this. Return the issues as a list and do the ul/li stuff in the html template instead.

Reply:
This validation error message would be a string "Remaining misconceptions that need to be addressed: misconception 1, misconception 2, misconception 3". Not sure how to parse this into in the HTML so that it displays:

Remaining misconceptions that need to be addressed:
- misconception 1
- misconception 2
- misconception 3

One way is to separate the validation error for this so that it returns a list of messages like:
misconception 1 needs to be addressed, misconception 2 needs to be addressed etc. In the HTML, we can simple determine if the message is a list and display it accordingly.

Reply:
Will there be other validation errors besides ones that are of this type?

If so, how will you display them in that case?

If not, why not just pass the list of misconception strings and have "Remaining misconceptions that need to be addressed" in the template?

Reply:
> Will there be other validation errors besides ones that are of this type? If so, how will you display them in that case?
Hm, don't think we have other types yet but I guess you're right that this might not scale well for other types in the future.

> If not, why not just pass the list of misconception strings and have "Remaining misconceptions that need to be addressed" in the template?
How do we determine that the validation message is related to misconceptions in the HTML?

Reply:
Well, if there are no other types, then won't the validation message always be related to unaddressed misconceptions? (Also not sure what you mean by "misconceptions in the html".)

If you want to you can make this clearer in the varname by using something like namesOfUnaddressedMisconceptions.

Reply:
Apologies for not being clear. Let me try again. This function is a question object validation function and it can return validation messages for a bunch of things. eg. `At least one answer should be marked correct`, `An interaction must be specified`, `At least 1 hint should be specfied`. So in the place where we display the validation message we currently don't have a way to differentiate between error messages and all of them are treated the same way (as strings).

Please correct me if I'm wrong but are you suggesting that we show `Remaining misconceptions that need to be addressed:` based on whether the returned value is a list and then render the list using li/ul in the HTML?

Reply:
No, not at all. In that case it seems like you do have validation errors that are of other types. So if you have those, as well as "unaddressed misconception" errors, how do you plan to show them?

E.g. let's say you have "At least 1 hint should be specified" and "Misconceptions A, B, C are not addressed". What does the displayed warning look like?

Reply:
We show only one message at a time.

Reply:
Ah, now I see the issue and how the above statements are consistent. Thanks!

In that case, I would suggest that you have two variables. One of them would hold the usual validation errors. The other would hold a list of unaddressed misconceptions.

If the first list is empty, show the second using bullet points. If the first list is not, then show the error(s) in the first list.

Does that work?

Reply:
Yeah, I can have validate() return something like this.
```ts
{
 validationError: XXX
 unaddressedMisonceptions: []
}
```

And use the condition you said above to render the bullet points.

Reply:
Yep, that's the rough idea. Can you use two separate functions though? We're trying not to pass dicts around in the codebase.

Reply:
Yeah that sounds like the way to go. ie. Separate out the unaddressed misconceptions into a separate validation function.

Reply:
Yup, exactly!

Btw -- shouldn't you be adding typedefs to the new code you're adding? I just realized. /cc @nishantwrp @bansalnitish @srijanreddy98

Reply:
I don't see typedefs used in this file. If there is a more organized task for adding typedefs for files and this is being tracked, I would prefer that this be handled as part of that. What do you think?

Reply:
I don't have an opinion -- please just check with @nishantwrp @bansalnitish @srijanreddy98 and get agreement on what's expected. If you all agree on a decision then I'm fine with it. Thanks!

Reply:
This `ObjectFactory` is not yet upgraded to angular. So, the types should be added later on when it is upgraded. Thanks.
/cc @bansalnitish @kevintab95 @srijanreddy98

Reply:
Thanks @nishantwrp!
Done.

Reply:
+1 with what Nishant suggested. We are adding types when we migrate stuff.